src/contract/assembled_transaction.ts

```diff
@@ -532,7 +571,10 @@ export class AssembledTransaction<T> {
 
   get result(): T {
     try {
-      return this.options.parseResultXdr(this.simulationData.result.retval);
+      if(!this.simulationData.result){
+        throw new Error("No simulation result!");
+      }
+      return this.options.parseResultXdr(this.simulationData.result?.retval);
```

Comment:
will this pop if `undefined` in the `?` case?

Reply:
That's actually checked on the line above, so we don't need the `?` check on this line. I can remove it.